examples/todo-mvc-react/src/todo.machine.ts

```diff
@@ -0,0 +1,135 @@
+import { createMachine, sendParent, EventFrom } from 'xstate';
+import { createModel } from 'xstate/lib/model';
+
+const todoModel = createModel(
+  {
+    id: '',
+    title: '',
+    prevTitle: '',
+    completed: false
+  },
+  {
+    events: {
+      TOGGLE_COMPLETE: () => ({}),
+      DELETE: () => ({}),
+      SET_COMPLETED: () => ({}),
+      SET_ACTIVE: () => ({}),
+      EDIT: () => ({}),
+      CHANGE: (value: string) => ({ value }),
+      COMMIT: () => ({}),
+      BLUR: () => ({}),
+      CANCEL: () => ({})
+    }
+  }
+);
+
+type TodoEvents = EventFrom<typeof todoModel>[keyof EventFrom<
+  typeof todoModel
+>];
+
```

Comment:
Is this probably temporary until `xstate/lib/model` settles down?

It was needed with `entry` transitions. I checked `exit` and `always` those require this type parameter too. (Did not tried delay transitions)